bblocks/import_tools/debt/wb_ids.py

```diff
@@ -0,0 +1,213 @@
+    """Read a json which contains all the IDS indicators"""
+    with open(BBPaths.debt_settings / f"{file}_indicators.json", "r") as fp:
+        return json.load(fp)
+
+
+@dataclass
+class DebtIDS(ImportData):
+    """Import data from the World Bank's International Debt Statistics database"""
+
+    def __post_init__(self):
+        """Set the path to the data folder and create it if it doesn't exist"""
+        self._path = BBPaths.raw_data / "ids_data"
+
+        # check the data path exists and if not create it
+        if not self._path.exists():
+            self._path.mkdir()
+
+    def _check_stored_data(
+        self, indicator: str, start_year: int, end_year: int
+    ) -> str | bool:
+        """Check if the data is already stored locally
+
+        This also checks if the years requested are inside another file.
+
+        Args:
+            indicator (str): The indicator to check
+            start_year (int): The start year of the data
+            end_year (int): The end year of the data
+
+        Returns:
+            str: The filename of the data if it exists
+            bool: False if the data doesn't exist
+
+        """
+
+        # get the filenames of all files in the path
+        files = [f.name for f in self._path.iterdir() if indicator in f.name]
+
+        for file in files:
+            # get the years from the filename
+            years = file.split("_")[-1].split(".")[0].split("-")
+
+            # if the requested data is inside an available file, return file name
+            if (int(years[0]) <= start_year) and (end_year <= int(years[1])):
+                return file
+
+        # if the data doesn't exist, return False
+        return False
+
+    @staticmethod
+    def _indicator_parameters(indicator: str) -> tuple[str, int, int]:
+        """Get the indicator, start year and end year from the indicator name."""
+        indicator, years = indicator.split("_")
+        start_year, end_year = years.split("-")
+        return indicator, int(start_year), int(end_year)
+
+    @classmethod
+    def get_available_indicators(cls) -> dict:
+        """Get a dictionary of all available indicators in the IDS database."""
+        return {k: v["name"] for k, v in read_indicators().items()}
+
+    @classmethod
+    def debt_service_indicators(cls, detailed_category: bool = True) -> dict:
+        """Get a dictionary of Debt Service indicators in the IDS database."""
+        return {
+            k: v["detailed_category" if detailed_category else "broad_category"]
+            for k, v in read_indicators("debt_service").items()
+        }
+
+    @classmethod
+    def debt_stocks_indicators(cls, detailed_category: bool = True) -> dict:
+        """Get a dictionary of Debt Service indicators in the IDS database."""
+        return {
+            k: v["detailed_category" if detailed_category else "broad_category"]
+            for k, v in read_indicators("debt_stocks").items()
+        }
```

Comment:
These class methods have similar functionality and implementation. Perhaps it is clearer to the user to have one function `get_available_indicators` passing in "all", "debt service" and "debt stock". Or perhaps rename to `get_debt_service_indicators` and `get_debt_stocks_indicators`

Reply:
> These class methods have similar functionality and implementation. Perhaps it is clearer to the user to have one function `get_available_indicators` passing in "all", "debt service" and "debt stock". Or perhaps rename to `get_debt_service_indicators` and `get_debt_stocks_indicators`

Single responsibility would suggest they should be separate functions. 
If they returned different views of the same data I would be more keen to make it a parameter. But in this case the data looks different.